hibernate-core/src/test/java/org/hibernate/orm/test/notfound/exception/NotFoundExceptionTest.java

```diff
@@ -0,0 +1,240 @@
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@DomainModel(
+		annotatedClasses = {
+				NotFoundExceptionTest.ChessGame.class,
+				NotFoundExceptionTest.ChessPlayer.class
+		}
+)
+@SessionFactory
+@TestForIssue(jiraKey = "HHH-15229")
+public class NotFoundExceptionTest {
+
+	@AfterEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session ->
+						session.createNativeMutationQuery( "delete from CHESS_GAME" ).executeUpdate()
+		);
+	}
+
+	@Test
+	public void testNotFoundIgnore(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session ->
+						session.createNativeMutationQuery(
+										"INSERT INTO CHESS_GAME(id, player_black_id) VALUES (1, 1)" )
+								.executeUpdate()
+
+		);
+
+		scope.inTransaction(
+				session -> {
+					ChessGame game = session.find( ChessGame.class, 1L );
+					assertNotNull( game, "Returned entity shouldn't be null" );
+					assertNull(
+							game.getPlayerBlack(),
+							"Broken foreign key reference with NotFoundAction.IGNORE should return null"
+					);
+					assertNull( game.getPlayerWhite() );
+				}
+		);
+
+		scope.inTransaction(
+				session -> {
+					ChessGame chessGame = session.getReference( ChessGame.class, 1L );
+					assertNotNull( chessGame );
+					assertNull(
+							chessGame.getPlayerBlack(),
+							"Broken foreign key reference with NotFoundAction.IGNORE should return null"
+					);
+					assertNull( chessGame.getPlayerWhite() );
+				}
+		);
+	}
+
+	@Test
+	public void testNotFoundIgnoreAndNotFoundException(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session ->
+						session.createNativeMutationQuery(
+										"INSERT INTO CHESS_GAME(id, player_white_id, player_black_id) VALUES (1, 1, 2)" )
+								.executeUpdate()
+		);
+
+		Assertions.assertThrows(
+				FetchNotFoundException.class, () -> scope.inTransaction(
+						session ->
+								session.find( ChessGame.class, 1L )
+				)
+		);
+
+		Assertions.assertThrows(
+				FetchNotFoundException.class, () -> scope.inTransaction(
+						session -> {
+							ChessGame chessGame = session.getReference( ChessGame.class, 1L );
+							assertNotNull( chessGame );
+							// this trigger ChessGame initialization, and because playerBlack cannot be found the FetchNotFoundException is thown
```

Comment:
super minor. this trigger -> this triggers, :)

Reply:
Fixed, thanks @NathanQingyangXu